### DISCUSSION TITLE
How do I handle rate limits between different instances/clusters?

### DISCUSSION THREAD

**Discussion by Luc1412** [Rapptz/discord.py#6395] [Q&A] [answered]
I created a multi-cluster bot with the help of https://github.com/XuaTheGrate/Discord-Clusters. I use Redis to sync data between all instances. 
Discord.py already provides a build-in rate limit handler, but this won't queue and sync through all instances.
What is the best practice of handling rate limits between all instances?

**Comment by Rapptz** [comment, marked as answer]
Rate limit buckets are per-guild and per-channel so separate guilds won't share those rate limits. 

The only complication comes with the global rate limit which is 50/s. However once you hit that you're kind of out of luck at the moment in the regular library since at best they'd all stop after hitting it but go back to doing it later.

I suppose in a future version there could be hooks into the global rate limit lock to allow someone to implement their own, similar to the identify hook in the client right now.
